notebooks/Composer_Functional.ipynb

```diff
@@ -0,0 +1,306 @@
+    "    # randaumgent and auxmix need PIL images, and so have to come\n",
+    "    # before ToTensor() in the list of transforms\n",
+    "    if 'randaug' in which_transforms:\n",
+    "        train_transforms.insert(0, cf.randaugment_image)\n",
+    "    if 'augmix' in which_transforms:\n",
+    "        train_transforms.insert(0, cf.augmix_image)\n",
+    "    # other augmentations can operate on batches of tensors, and so\n",
+    "    # can come after ToTensor()\n",
+    "    if 'colout' in which_transforms:\n",
+    "        train_transforms.append(cf.colout_batch)\n",
+    "    if 'cutout' in which_transforms:\n",
+    "        train_transforms.append(cf.cutout_batch)\n",
+    "\n",
+    "    test_transform = transforms.Compose(shared_transforms)\n",
+    "    train_transform = transforms.Compose(train_transforms)\n",
+    "\n",
+    "    trainset = torchvision.datasets.CIFAR10(root=datadir, train=True,\n",
+    "                                            download=True, transform=train_transform)\n",
+    "    trainloader = torch.utils.data.DataLoader(trainset, batch_size=batch_size,\n",
+    "                                            shuffle=True, num_workers=2)\n",
+    "\n",
+    "    testset = torchvision.datasets.CIFAR10(root=datadir, train=False,\n",
+    "                                           download=True, transform=test_transform)\n",
+    "    testloader = torch.utils.data.DataLoader(testset, batch_size=batch_size,\n",
+    "                                             shuffle=False, num_workers=2)\n",
+    "    \n",
+    "    return trainloader, testloader"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's see how our model does with just these changes."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model = Net()\n",
+    "# only use one data augmentation since our small model runs quickly\n",
+    "# and allows the dataloader little time to do anything fancy\n",
+    "trainloader, testloader = make_augmented_dataloaders(['colout'])\n",
+    "train_and_eval(model, trainloader, testloader)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As we might expect, adding data augmentation doesn't help us when we aren't training long enough to start overfitting.\n",
+    "\n",
+    "Now let's try using some algorithms that modify the model. We're going to keep things simple and just add a [Squeeze-and-Excitation](https://docs.mosaicml.com/en/latest/method_cards/squeeze_excite.html) module after the larger of the two conv2d operations in our model. We show examples of how to use other algorithms as well as a reference."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def modify_model(model: torch.nn.Module, which_algorithms: Sequence[str]):\n",
```

Comment:
same thing.. maybe better to un-functionify things so its more interactive.

Reply:
done